sources/api/apiserver/src/server/mod.rs

```diff
@@ -749,6 +822,13 @@ fn transaction_name(query: &web::Query<HashMap<String, String>>) -> &str {
     query.get("tx").map(String::as_str).unwrap_or("default")
 }
 
+fn strength(query: &web::Query<HashMap<String, String>>) -> Strength {
+    match query.get("strength") {
+        Some(strength) => strength.parse::<Strength>().unwrap_or(Strength::Strong),
+        None => Strength::Strong,
+    }
+}
+
```

Comment:
If the given query string isn't `strong` or `weak`, we should probably throw an error.

```suggestion
fn strength(query: &web::Query<HashMap<String, String>>) -> Result<Strength> {
    Ok(query.get("strength")
        .map(str::parse::<Strength>)
        .transpose()
        .context(SomethingSomethingSomething)?
        .unwrap_or_default())
}
```